src/UID2.Client/TokenHelper.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace UID2.Client
+{
+    internal class TokenHelper
+    {
+        private Uid2ClientHelper _uid2ClientHelper;
+        private KeyContainer _container;
+
+        internal TokenHelper(string endpoint, string authKey, string secretKey)
+        {
+            _uid2ClientHelper = new Uid2ClientHelper(endpoint, authKey, secretKey);
+        }
+
+        internal DecryptionResponse Decrypt(string token, DateTime now, string domainNameFromBidRequest, ClientType clientType)
+        {
+            var container = Volatile.Read(ref _container);
+            if (container == null)
+            {
+                return DecryptionResponse.MakeError(DecryptionStatus.NotInitialized);
+            }
+
+            if (!container.IsValid(now))
+            {
+                return DecryptionResponse.MakeError(DecryptionStatus.KeysNotSynced);
+            }
+
+            try
+            {
+                return UID2Encryption.Decrypt(token, container, now, domainNameFromBidRequest, container.GetIdentityScope(), clientType);
+            }
+            catch (Exception)
+            {
+                return DecryptionResponse.MakeError(DecryptionStatus.InvalidPayload);
+            }
+        }
+        public EncryptionDataResponse Encrypt(string rawUid)
+        {
+            return Encrypt(rawUid, DateTime.UtcNow);
+        }
+
+        internal EncryptionDataResponse Encrypt(string rawUid, DateTime now)
+        {
+            var container = Volatile.Read(ref _container);
+            if (container == null)
+            {
+                return EncryptionDataResponse.MakeError(EncryptionStatus.NotInitialized);
+            }
+            else if (!container.IsValid(now))
+            {
+                return EncryptionDataResponse.MakeError(EncryptionStatus.KeysNotSynced);
+            }
+
+            return UID2Encryption.Encrypt(rawUid, container, container.GetIdentityScope(), now);
+        }
+
+        internal RefreshResponse Refresh()
+        {
+            return RefreshInternal().Result;
```

Comment:
This could result in [deadlock](https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/async-scenarios#important-info-and-advice)

Reply:
This is old code from https://github.com/IABTechLab/uid2-client-net/blob/main/src/UID2.Client/UID2Client.cs#L114 - we'll have to address this another time. It's called once per sec per bidder and I haven't heard of any deadlocks :/

Reply:
created UID2-2946